back/engines/commercial/content_builder/app/policies/content_builder/layout_image_policy.rb

```diff
@@ -0,0 +1,18 @@
+module ContentBuilder
+  class LayoutImagePolicy < ApplicationPolicy
+    def create?
+      # The content buildable is not provided and
+      # so we cannot check if the user moderates
+      # the corresponding context. Instead, we
+      # check if the user has any moderation
+      # rights.
+      user&.active? && user_role_service.moderates_something?(user)
+    end
+
+    private
+
+    def user_role_service
+      @user_role_service ||= UserRoleService.new
```

Comment:
`user_role_service` is a private method and it is only invoked once in this class, so I think the memoization is not required. One could argue that `create?` maybe called multiple times. That is true, but is it a use case? And even if it is a use case, would not memoizing `user_role_service` be a performance issue?

Reply:
That is a good point. The test sets expectations on a method on the result of a private method, because dependency injection is not possible (the `user_role_service` cannot be set from outside the class). That is not ideal, because the spec relies on a private implementation detail, but on the other hand the usage of the `user_role_service` is a very important aspect of the described behaviour. We could consider allowing dependency injection (by passing an optional `UserRoleService` to `initialize` for instance). When I wrote the specs for the layout policy, I considered that, but I thought it would go too far. But I think we should reconsider at some point, because as you indicate, the memoization is required for the spec to function correctly. I would not do that as part of this PR. Let's keep it in mind, discuss this later, and improve the code and tests together. In the meantime, we can keep this code.